packages/core/src/slider/create-slider-state.ts

```diff
@@ -121,11 +121,17 @@ export function createSliderState(props: StateOpts): SliderState {
   };
 
   const getThumbMinValue = (index: number) => {
-    return index === 0 ? props.minValue!() : values()[index - 1];
+    return index === 0
+      ? props.minValue!()
+      : values()[index - 1] +
+          props.minStepsBetweenThumbs!() * props.step!() * (values().length - 1);
   };
 
   const getThumbMaxValue = (index: number) => {
-    return index === values().length - 1 ? props.maxValue!() : values()[index + 1];
+    return index === values().length - 1
+      ? props.maxValue!()
+      : values()[index + 1] -
+          props.minStepsBetweenThumbs!() * props.step!() * (values().length - 1);
```

Comment:
Adding the * (values().length - 1) seems to not create the correct result. I tested this with 3 thumbs, selecting the middle thumb, and using the Home and End keys to set the thumb to it's min/max values.
```suggestion
          props.minStepsBetweenThumbs!() * props.step!()
```

Reply:
I think if you can change the `getThumbMinValue` and `getThumbMaxValue` as suggested above or let me know if I am misunderstanding something, it looks good to me. 

Regarding your last comment, I hadn't noticed that issue. Feel free to include it in the PR as well.

Thanks so much for the PR @royeden!